src/main.c

```diff
@@ -64,33 +67,47 @@ int main(int argc, char *argv[]) {
    rsLog("This is free software, and you are welcome to redistribute it");
    rsLog("under certain conditions; see COPYING for details.");
 
+   pthread_t athread;
+   
    RSConfig config;
    RSSystem system;
    RSCompress compress;
    RSBufferCircle circle;
    RSOutput output = {0};
+   RSAudio audio;
    rsConfigLoad(&config);
    // We have to call this one first since it might change the config width/height
    rsXlibSystemCreate(&system, &config);
    rsCompressCreate(&compress, &config);
    size_t capacity = (size_t)(config.duration * config.framerate);
    rsBufferCircleCreate(&circle, capacity);
+   rsAudioCreate(&audio, &config);
+
+   for (;;) {
```

Comment:
Can you keep the original loop? Just have the output take a second buffer and the audio implementation, if it uses a thread, to make a copy of its buffer pass over.